cassandra/src/main/scala/geotrellis/spark/io/cassandra/CassandraIndexing.scala

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2016 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
@pomadchin ,

I squeezed in a bit more time to work on this today and got some of the duplicate code cleaned up and pulled into this `CassandraIndexing` class.
 
I'm going to try to dive into some sort of benchmarking thing to see if I can prove to myself whether or not this indexing change really has done anything to improve read speed next. 

Thanks,
Andy